File: data_structures_js/mystack.js
class MyStack {
	constructor(max=10) {
		this.max = max;
		this.arr = [];
		this.top = -1;
	}
	isEmpty() { return (this.top === -1); }
	isFull() { return (this.top === this.max-1); }
	push(item) {
		if (!this.isFull()) {
			this.arr.push(item)
			this.top++
		}
	}
	pop() {
		if (!this.isEmpty()) {
			this.top--
			return this.arr.pop();
		}
	}
	printStack() {
		for (let i=0; i<this.arr.length; i++) {
			console.log(this.arr[i])
		}
		console.log("-----")
	}
}

if (require.main === module) {
	ms = new MyStack();
	ms.push(10);
	ms.push(15);
	ms.push(20);
	ms.printStack();
	ms.pop();
	ms.pop();
	ms.printStack();
}

File: data_structures_cpp/mystack.cpp
#include <iostream>
using namespace std;

template <class T = int>
class MyStack {
	public:
		int max;
		T *arr;
		int top;

		MyStack () {
			max = 10;
			arr = new T[max];
			top = -1;
		}

		int isFull() {
			if (top == max-1) { return 1; }
			return 0;
		}

		int isEmpty() {
			if (top == -1) { return 1; }
			return 0;
		}

		void push(T n) {
			if (!isFull()) { arr[ ++top ] = n; }
		}

		T pop() {
			if (!isEmpty()) { return arr[top--]; }
		}

		void printStack() {
			cout << "Printing Stacks: " << endl;
			for (int i=0; i<=top; i++) {
				cout << arr[i] << endl;
			}
		}
};

int main(){
	MyStack <char> ms;
	cout << "isFull: " << ms.isFull() << endl;
	cout << "isEmpty: " << ms.isEmpty() << endl;

	ms.push('a');
	ms.push('b');
	ms.push('c');
	ms.push('d');

	ms.printStack();
	ms.pop();
	ms.pop();
	ms.pop();
	ms.pop();
	ms.printStack();

	return 0;
}